### DISCUSSION TITLE
Access string in sml array

### DISCUSSION THREAD

**Discussion by iot-sle** [arendst/Tasmota#15200] [Q&A] [answered]
>M 2
+1,16,s,0,9600,ESY-SWS
1,77070100000009ff@#,*,,ID,0 ;1

sml[1] should contain "09014553591103b90fc6". The meter is exporting the ID as a string.

MQTT is reporting "ESY-SWS":{"ID":"09014553591103b90fc6"} correctly. But when I try to access the value no success:
What I tried:

print %sml[1]%
S_ID=sml[1]
print %S_ID%

and in >W section 

%S_ID% and sml[1]

using sml[1] in >W  is always showing 0. Any idea how to access the string correctly?

**Comment by gemu2015** [comment, marked as answer]
smls[m]   m = meter number, returns the meter id string (decoded with #)

**Reply by nikito7** [reply]
this work with modbus (mN1) ?

**Reply by gemu2015** [reply]
yes, it should

**Comment by iot-sle** [comment]
Thank you.
